heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -116,11 +134,65 @@ export default class LintPlugin implements IHeftTaskPlugin<ILintPluginOptions> {
         // Warn since don't run the linters when in watch mode.
         taskSession.logger.terminal.writeWarningLine("Linting isn't currently supported in watch mode");
       } else {
+        if (!inTypescriptPhase) {
+          const fix: boolean = this._checkFix(taskSession, pluginOptions);
+          const sarifLogPath: string | undefined = this._getSarifLogPath(heftConfiguration, pluginOptions);
+          // If we are not in the typescript phase, we need to create a typescript program
+          // from the tsconfig file
+          const tsProgram: IExtendedProgram = await this._createTypescriptProgramAsync(
+            heftConfiguration,
+            taskSession
+          );
+          const rootFiles: readonly string[] = tsProgram.getRootFileNames();
+          const changedFiles: Set<IExtendedSourceFile> = new Set<IExtendedSourceFile>();
+          rootFiles.forEach((rootFilePath: string) => {
+            const sourceFile: TTypescript.SourceFile | undefined = tsProgram.getSourceFile(rootFilePath);
+            changedFiles.add(sourceFile as IExtendedSourceFile);
+          });
+
+          const lintingPromise: Promise<void> = this._lintAsync({
+            taskSession,
+            heftConfiguration,
+            fix,
+            sarifLogPath,
+            tsProgram,
+            changedFiles
+          });
+          lintingPromise.catch(() => {
+            // Suppress unhandled promise rejection error
+          });
+          // Hold on to the original promise, which will throw in the run hook if it unexpectedly fails
+          this._lintingPromises.push(lintingPromise);
```

Comment:
We can just await `this._lintAsync` directly; the shenanigans with `this._lintingPromises` are only needed if we are running triggered off of the typescript task.

Reply:
Modified in [772829d](https://github.com/microsoft/rushstack/pull/5239/commits/772829d703d4a0052d99a0b0e2a82b6d5bcc6baf)